prowler/providers/aws/aws_provider.py

```diff
@@ -1030,24 +1044,62 @@ def test_connection(
                 is_connected=True,
             )
 
-        except (ClientError, ProfileNotFound, NoCredentialsError) as credentials_error:
+        except AWSSetUpSessionError as setup_session_error:
             logger.error(
-                f"{credentials_error.__class__.__name__}[{credentials_error.__traceback__.tb_lineno}]: {credentials_error}"
+                f"AWSSetUpSessionError[{setup_session_error.__traceback__.tb_lineno}]: {setup_session_error}"
             )
             if raise_on_exception:
-                raise credentials_error
+                raise setup_session_error
+            return Connection(error=setup_session_error)
 
-            return Connection(error=credentials_error)
-
-        except ArgumentTypeError as validation_error:
+        except AWSArgumentTypeValidationError as validation_error:
             logger.error(
-                f"{validation_error.__class__.__name__}[{validation_error.__traceback__.tb_lineno}]: {validation_error}"
+                f"AWSArgumentTypeValidationError[{validation_error.__traceback__.tb_lineno}]: {validation_error}"
             )
             if raise_on_exception:
                 raise validation_error
-
             return Connection(error=validation_error)
 
+        except AWSIAMRoleARNRegionNotEmtpy as error:
```

Comment:
Why are you adding this here?

Reply:
But this one is just of a few that can be raised, we should add the rest too, right?

Reply:
You are right! Done